src/context/AppProvider/AppContext.tsx

```diff
@@ -142,91 +122,37 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
-              if (!supportsBTC(wallet)) continue
-
-              const utxoAdapter = adapter as unknown as UtxoBaseAdapter<UtxoChainId>
-
-              for (const accountType of utxoAdapter.getSupportedAccountTypes()) {
-                const { bip44Params, scriptType } = utxoAccountParams(chainId, accountType, 0)
-                const pubkeys = await wallet.getPublicKeys([
-                  {
-                    coin: utxoAdapter.getCoinName(),
-                    addressNList: bip32ToAddressNList(toRootDerivationPath(bip44Params)),
-                    curve: 'secp256k1',
-                    scriptType,
-                  },
-                ])
-
-                if (!pubkeys?.[0]?.xpub) throw new Error('failed to get public key')
-
-                const pubkey = convertXpubVersion(pubkeys[0].xpub, accountType)
-                if (!pubkey) continue
-
-                acc.push({ [chainId]: pubkey })
-                const accountId = toAccountId({ chainId, account: pubkey })
-                accMeta[accountId] = { bip44Params, accountType }
-              }
-              break
-            }
-
-            case CHAIN_NAMESPACE.Evm: {
-              if (chainReference === CHAIN_REFERENCE.EthereumMainnet) {
-                if (!supportsETH(wallet)) continue
-              }
-              if (chainReference === CHAIN_REFERENCE.AvalancheCChain) {
-                if (!supportsEthSwitchChain(wallet)) continue
-              }
-
-              const bip44Params = adapter.getBIP44Params({ accountNumber: 0 })
-              const pubkey = await adapter.getAddress({ bip44Params, wallet })
-              if (!pubkey) continue
-              const accountId = toAccountId({ chainId, account: pubkey.toLowerCase() })
-              acc.push({ [chainId]: pubkey.toLowerCase() })
-              accMeta[accountId] = { bip44Params }
-              break
-            }
-            case CHAIN_NAMESPACE.CosmosSdk: {
-              if (chainReference === CHAIN_REFERENCE.CosmosHubMainnet) {
-                if (!supportsCosmos(wallet)) continue
-              }
-              if (chainReference === CHAIN_REFERENCE.OsmosisMainnet) {
-                if (!supportsOsmosis(wallet)) continue
-              }
-
-              const bip44Params = adapter.getBIP44Params({ accountNumber: 0 })
-              const pubkey = await adapter.getAddress({ bip44Params, wallet })
-              if (!pubkey) continue
-              acc.push({ [chainId]: pubkey })
-              const accountId = toAccountId({ chainId, account: pubkey })
-              accMeta[accountId] = { bip44Params }
-              break
-            }
-            default:
-              break
-          }
-        }
-
-        dispatch(accountSpecifiers.actions.setAccountSpecifiers(acc))
-        dispatch(portfolio.actions.setAccountMetadata(accMeta))
+        const accountNumber = 0
+        const chainIds = supportedChains
+        const accountIdsAndMetadata = await deriveAccountIdsAndMetadata({
```

Comment:
guts of the PR - this function can now be called from elsewhere (add account modal) with the next account number and accountId, using the return type to dispatch and upsert with additional accounts